src/filesystem_helper.cpp

```diff
@@ -341,6 +343,38 @@ path create_temp_directory(const std::string & base_name, const path & parent_pa
   return final_path;
 }
 
+std::filesystem::path create_temporary_directory(
+  const std::string & base_name, const std::filesystem::path & parent_path, size_t max_tries)
+{
+  auto it =
+    std::find(base_name.begin(), base_name.end(), std::filesystem::path::preferred_separator);
+  if (it != base_name.end()) {
+    throw std::invalid_argument("The base_name contain directory-separator");
+  }
+  // mersenne twister random generator engine seeded with the std::random_device
+  std::mt19937 random_generator(std::random_device{}());
+  std::uniform_int_distribution<> distribution(0, 999999);
```

Comment:
I think this interval is too small, now that I look at it.  When I ran some tests locally, I always got `0` as the first digit.  And that makes sense, because 999999 == F423F hex, so there will never be something in the first digit.  For 6 digits of hex, we should make this 0xFFFFFF:

```suggestion
  std::uniform_int_distribution<> distribution(0, 0xFFFFFF);
```

Reply:
@clalancette Good catch! I forgot that we convert in HEX and that in hex the range should be bigger.
Fixed in the https://github.com/ros2/rcpputils/pull/198/commits/8b4087d95df458948f20f0d533c56330d5733606 commit.